vllm_spyre/v1/worker/spyre_model_runner.py

```diff
@@ -715,8 +723,52 @@ def __init__(
             vocab_size=vllm_config.model_config.get_vocab_size(),
         )
 
-    def _set_free_blocks(self, num_blocks: int) -> None:
-        self.free_blocks = deque([i for i in range(num_blocks)])
+    def _set_blocks(self, num_blocks: int) -> None:
+        self.n_blocks = num_blocks
+        self.block_pool = deque([i for i in range(self.n_blocks)])
+
+    def _get_num_blocks_available(self) -> int:
+        """Function returns the number of available blocks/pages.
+        Will eventually contain a function in torch_sendnn which reads 
+        the actual value provided by the compiler for backend sendnn"""
+
```

Comment:
can we consolidate some of the `if envs_spyre.VLLM_SPYRE_N_BLOCKS > 0:` checks into here and return the override from this method?